ReusableViews/ReusableBtn.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  ReusableBtn.swift
+//  WeatherDemoApp
+//
+//  Created by ahmed on 03/04/2024.
+//
+
+import UIKit
+
+final class ReusableButton: UIButton {
```

Comment:
Resuable name is very generic and should be avoided. You can used names that represent the button shape like TextButton - FilledButton - RoundedButton - ..etc.